pkg/cache/application.go

```diff
@@ -273,42 +312,46 @@ func (app *Application) Schedule() {
 			log.Logger().Warn("failed to handle SUBMIT app event",
 				zap.Error(err))
 		}
-	case states.Accepted:
-		ev := NewRunApplicationEvent(app.GetApplicationID())
-		if err := app.handle(ev); err != nil {
-			log.Logger().Warn("failed to handle RUN app event",
-				zap.Error(err))
-		}
+	case states.Reserving:
```

Comment:
Good question. The Accepted part is moved to

```
func (app *Application) postAppAccepted(event *fsm.Event) {
	// if app has taskGroups defined, it goes to the Reserving state before getting to Running
	var ev events.SchedulingEvent
	if len(app.taskGroups) != 0 {
		ev = NewSimpleApplicationEvent(app.applicationID, events.TryReserve)
		dispatcher.Dispatch(ev)
	} else {
		ev = NewRunApplicationEvent(app.applicationID)
	}
	dispatcher.Dispatch(ev)
}
```

it checks if an app has taskGroups, if not, directly transit the app to Running state; otherwise it will be going to the Reserving state. This is due to the state machine graph change, see more [this doc](https://docs.google.com/document/d/1P-g4plXIJ9Xybp-jyKySI18P3rkGQPuTutGYhv1LaQ8/edit#heading=h.laq9umr6wnyj).